src/averageAges.js

```diff
@@ -55,7 +75,22 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const childrenWithMotherHere = onlyWithSon
+    ? people.filter((child, index, currentList) => (
+      child.sex === 'm'
+      && currentList.some(person => person.name === child.mother)
+    ))
+    : people.filter((child, index, currentList) => (
+      currentList.some(person => person.name === child.mother)
+    ));
+
+  const ageDiffs = childrenWithMotherHere.map(child => (
```

Comment:
```suggestion
  const ageDiffs = motherAge.map(child => (
```
Very long name for variable, there is no need to use it

Reply:
What is `motherAge`?
`childrenWithMotherHere` stores children whose mothers are stored in the list